spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -310,39 +310,31 @@ def add_devices(recording: se.RecordingExtractor, nwbfile=None,
             Should be of the format
                 metadata['Ecephys']['Device'] = [{'name': my_name,
                                                   'description': my_description}, ...]
+
+        Missing keys in an element of metadata['Ecephys']['Device'] will be auto-populated with defaults.
         '''
         if nwbfile is not None:
             assert isinstance(nwbfile, NWBFile), "'nwbfile' should be of type pynwb.NWBFile"
         defaults = {'name': 'Device',
                     'description': 'no description'}
 
         if metadata is None:
-            metadata = dict()
+            metadata = {}
+            metadata['Ecephys'] = {}
+            metadata['Ecephys']['Device'] = []
 
-        if len(metadata.keys()) > 0:
-            if 'Ecephys' in metadata \
-                    and 'Device' in metadata['Ecephys'] \
-                    and type(metadata['Ecephys']['Device']) is list \
-                    and metadata['Ecephys']['Device']:
-                for j, dev in enumerate(metadata['Ecephys']['Device']):
-                    if type(dev) is dict:
-                        # Will not overwrite the named device if already in nwbfile
-                        if dev.get('name', defaults['name']) not in nwbfile.devices:
-                            device_kwargs = fill_kwargs_from_defaults(defaults, dev)
-                            nwbfile.create_device(**device_kwargs)
-                    else:
-                        print(f"Warning: Expected metadata['Ecephys']['Device'][{j}] to be"
-                              " a dictionary with keys 'name' and 'description'!"
-                              f" Device [{j}] will not be created.")
-            else:
-                print("Warning: metadata must be a list of dictionaries of the form"
-                      " metadata['Ecephys']['Device'] = [{'name': my_name,"
-                      " 'description': my_description}, ...]!")
+        assert len(metadata.keys()) > 0, "Expected metadata must be a non-empty dictionary!"
+        assert 'Ecephys' in metadata, "Expected metadata to have key 'Ecephys'!"
+        assert 'Device' in metadata['Ecephys'], "Expected metadata['Ecephys'] to have key 'Device'!"
+        assert type(metadata['Ecephys']['Device']) is list, "Expected metadata['Ecephys']['Device'] to be a list!"
+
+        for j, dev in enumerate(metadata['Ecephys']['Device']):
+            assert type(dev) is dict, f"Expected metadata['Ecephys']['Device'][{j}] to be a dictionary!"
+            if dev.get('name', defaults['name']) not in nwbfile.devices:
+                nwbfile.create_device(**dict(defaults, **dev))
```

Comment:
```suggestion
        assert isinstance(x, dict) for x in metadata['Ecephys']['Device'], "Expected metadata['Ecephys']['Device'] to be a list of dicts"

        for dev in metadata['Ecephys']['Device']:
            if dev.get('name', defaults['name']) not in nwbfile.devices:
                nwbfile.create_device(**dict(defaults, **dev))
```

Reply:
This single assertion will do the work of all of them

Reply:
`isinstance` should be used instead of `type(x) == Type`

Reply:
All occurrences of `type `ought to now be replaced by `isinstance`